File: 백준/Gold/2470. 두 용액/두 용액.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        String[] input = br.readLine().split(" ");
        long[] arr = new long[n];
        for (int i = 0; i < n; i++) {
            arr[i] = Long.parseLong(input[i]);
        }

        Arrays.sort(arr);

        long min = Long.MAX_VALUE, ansLeft = 0, ansRight = 0;
        int left = 0, right = n - 1;
        while (left < right) {
            long temp = arr[left] + arr[right];

            if (Math.abs(temp) < min) {
                min = Math.abs(temp);
                ansLeft = arr[left];
                ansRight = arr[right];
            }

            if (temp < 0) {
                left++;
            } else {
                right--;
            }
        }

        System.out.println(ansLeft + " " + ansRight);
    }
}

File: 백준/Gold/2470. 두 용액/두 용액.js
let [n, input] = require("fs")
  .readFileSync(process.platform === "linux" ? "/dev/stdin" : "input.txt")
  .toString()
  .trim()
  .split("\n");

n = Number(n);
let arr = input
  .split(" ")
  .map(Number)
  .sort((a, b) => a - b);

let min = Number.MAX_SAFE_INTEGER;
let left = 0,
  right = n - 1,
  ansLeft = 0,
  ansRight = 0;
while (left < right) {
  let temp = arr[left] + arr[right];

  if (Math.abs(temp) < min) {
    min = Math.abs(temp);
    ansLeft = arr[left];
    ansRight = arr[right];
  }

  if (temp < 0) {
    left++;
  } else {
    right--;
  }
}

console.log(ansLeft, ansRight);
